### DISCUSSION TITLE
How to redirect / route to list page after create

### DISCUSSION THREAD

**Discussion by sonamlepcha2010** [filamentphp/filament#1153] [Help]
Hello everyone. I am new to Laravel as well as to Filament. I am trying to create a small project and I have a small query. After the record is created I would like to route/redirect to the list page rather than edit page. I tried multiple ways but ended up either getting errors, redirected to edit page or end up in the same page. Any help would be much appreciated. Thanks

**Comment by danharrin** [comment]
Try this, on the Create page class:

```
protected function getRedirectUrl(): ?string
{
    return static::getResource()::getUrl('index');
}
```

**Comment by sonamlepcha2010** [comment]
It worked. Thank you very much.

**Comment by sergioraldana** [comment]
Hi, I tried but I got this:

`Declaration of App\Filament\Resources\RolResource\Pages\CreateRol::getRedirectUrl(): ?string must be compatible with Filament\Resources\Pages\CreateRecord::getRedirectUrl(): string`

**Reply by danharrin** [reply]
The error tells you what is wrong, the return type is different

**Reply by faridsa** [reply]
@danharrin This solution works pretty well, but I'm wondering if there is a global setting for that, instead of adding this function to all create and edit pages. 
Thank you!

**Reply by masumcity** [reply]
try this

protected function getRedirectUrl(): string
    {
        return $this->getResource()::getUrl('index');
    }

**Reply by Andrewsuares** [reply]
Works great! Thanks!!

**Comment by internetbug256** [comment]
> @danharrin This solution works pretty well, but I'm wondering if there is a global setting for that, instead of adding this function to all create and edit pages. Thank you!

Hi. I am looking for exactly the same. I found lots of defaults for all my resources I would like to customize. I know there is a place to customize this. Some boot method somewhere...

**Reply by leandrocfe** [reply]
Maybe a trait?

```php
trait MyTrait
{
    protected function getRedirectUrl(): string
    {
        return $this->getResource()::getUrl('index');
    }
}
```

```php
class MyCreatePage extends CreateRecord
{
    use MyTrait;
    ...
}
```

**Reply by internetbug256** [reply]
Yeah, I came to the global approach last night. Was actually a simple PHP way: just extend the CreateRecord and EditRecord Filament classes. That worked perfectly

**Comment by joao-antonio-gomes** [comment]
Send to edit page:

```php
protected function getRedirectUrl(): string {
    return static::getResource()::getUrl('edit', [
        'record' => $this->record
    ]);
}
```

**Reply by alisalehi1380** [reply]
tnks @joao-antonio-gomes  ✌️❤️

**Comment by pratikkuikel** [comment]
Any way to set global redirect for createRecord using serviceProvider ?

**Reply by intrepidws** [reply]
Did you ever figure this out? I'd love to be able to do this so that the app will always attempt to redirect to the edit page first, before trying the view page. To me, the default logic is very much backwards.

**Comment by devsumitkarn** [comment]
Method 'App\Filament\Resources\TableResource\Pages\CreateTable::getRedirectUrl()' is not compatible with method

**Reply by wahidikqbal** [reply]
protected function getRedirectUrl(): string
    {
        return $this->getResource()::getUrl('index');
    }

**Comment by alisalehi1380** [comment]
If you are on the edit page, do this.
```php
protected function getRedirectUrl(): ?string
{
    if ($this->previousUrl === null) {
        return static::getResource()::getUrl();
    }

    return $this->previousUrl;
}
```
